event-publisher-spi/src/main/scala/uk/gov/nationalarchives/eventpublisherspi/EventPublisherProvider.scala

```diff
@@ -0,0 +1,67 @@
+package uk.gov.nationalarchives.eventpublisherspi
+
+import java.net.URI
+
+import io.circe.syntax.EncoderOps
+import org.keycloak.events.admin.AdminEvent
+import org.keycloak.events.{Event, EventListenerProvider}
+import org.keycloak.models.KeycloakSession
+import software.amazon.awssdk.http.apache.ApacheHttpClient
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sns.SnsClient
+import uk.gov.nationalarchives.aws.utils.SNSUtils
+import uk.gov.nationalarchives.eventpublisherspi.EventPublisherProvider.{EventDetails, EventPublisherConfig}
+import uk.gov.nationalarchives.eventpublisherspi.helpers.EventUtils.{AdminEventUtils, eventDetailsEncoder}
```

Comment:
Similar to the comment below, can you use `import io.circe.generic.auto._` instead of explicitly importing the encoder?